File: 01-Languages/Go/05-Control-Flow/Solutions/main.go
// main.go - FizzBuzz with if/else if and Go's single for loop keyword.
package main

import (
	"fmt"
	"strconv"
)

func fizzBuzz(n int) string {
	if n%15 == 0 {
		return "FizzBuzz"
	}
	if n%3 == 0 {
		return "Fizz"
	}
	if n%5 == 0 {
		return "Buzz"
	}
	return strconv.Itoa(n)
}

func main() {
	for i := 1; i <= 15; i++ {
		fmt.Println(fizzBuzz(i))
	}
}

File: 01-Languages/Rust/05-Control-Flow/Solutions/main.rs
// main.rs - FizzBuzz using a tuple pattern match on (n % 3, n % 5).

fn fizzbuzz(n: u32) -> String {
    match (n % 3, n % 5) {
        (0, 0) => "FizzBuzz".to_string(),
        (0, _) => "Fizz".to_string(),
        (_, 0) => "Buzz".to_string(),
        _ => n.to_string(),
    }
}

fn main() {
    for i in 1..=15 {
        println!("{}", fizzbuzz(i));
    }
}
